From b0d2aaa54ad934115f6abd54da81d9ecaa40ac1f Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 18 Apr 2021 22:07:39 +0300
Subject: [PATCH] [SW-39246] gaudi: erase rma paritition

Erase RMA partition, as a request from host
given in fit. This is required as a mechanism
to delete this partition in various cases,
for example: doing a fix and returning the product
to customer, finding RMA bug, etc.

Change-Id: I4bfe62b4331995b3140cfb31fa47d5b7c94d59a9
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/gaudi/rma_info.c b/gaudi/rma_info.c
index c6ccf15..d9d98a5 100644
--- a/gaudi/rma_info.c
+++ b/gaudi/rma_info.c
@@ -62,6 +62,20 @@
 	.flash_size = { FLASH_SECTOR_SIZE, FLASH_SECTOR_SIZE },
 };
 
+int rma_erase(void)
+{
+	int ret, i;
+
+	for (i = 0; i < ARRAY_SIZE(flash_mem_context.base); i++) {
+		ret = spi_erase(flash_mem_context.base[i],
+				flash_mem_context.flash_size[i]);
+		if (ret)
+			return ret;
+	}
+
+	return 0;
+}
+
 static int flash_read(size_t offset, void *data, size_t size)
 {
 	int flash_select = -1;
diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 15f06bb..86cac8c 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -151,6 +151,16 @@
 	return err;
 }
 
+__weak int rma_erase(void)
+{
+	return 1;
+}
+
+static int module_fwupd_load_type_rma_erase(void)
+{
+	return rma_erase();
+}
+
 static int module_fwupd_load_type_run(void)
 {
 	is_fit_checked = !fit_load_images();
@@ -172,6 +182,8 @@
 	switch (load_type) {
 	case FIT_LOAD_TYPE_NIC_EN_MASK:
 		return module_fwupd_load_type_nic_en(fit);
+	case FIT_LOAD_TYPE_RMA_ERASE:
+		return module_fwupd_load_type_rma_erase();
 	case FIT_LOAD_TYPE_RUN:
 		return module_fwupd_load_type_run();
 	default:
